### DISCUSSION TITLE
Total Project lines of code at a specific commit

### DISCUSSION THREAD

**Discussion by ErnestFistozz** [community/community#45955] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I am trying to track the total source code lines over time for a repository. is it possible to get the total project lines of code at a specific commit NOT only the added or deleted BUT the ENTIRE project LOC for a specific commit such that you can get a sense of how the repository's LOC behaves over a period of time.

The following API gives only returns lines of code related to that commit i.e it gives lines added, deleted and the total of the former two:
[https://api.github.com/repos/{repoOwnerId}/{repoId}/commits/{commitId}](https://api.github.com/repos/%7BrepoOwnerId%7D/%7BrepoId%7D/commits/%7BcommitId%7D)

Take for example:

`https://api.github.com/repos/apache/beam/commits/3c623abd631b493c8a71d05fe591065754c75d0c`

it will return 
`....
"stats": {
"total": 31,
"additions": 28,
"deletions": 3
}
....`

Obviously, these are simply the changes made for that commit, but I want the whole system LOC at the point ( taking into account the LOC for the files which were not changed).

Does GitHub have an API for that?

Any suggestion is welcome.

**Comment by jge162** [comment, marked as answer]
GitHub API does not provide the total lines of code (LOC) for a specific commit in a repository. You would need to periodically run the same process for multiple commits and record the results in a spreadsheet or similar tool.

**Reply by ErnestFistozz** [reply]
Thanks @jge162 .
